jscomp/frontend/bs_builtin_ppx.ml

```diff
@@ -214,6 +221,19 @@ let expr_mapper ~async_context ~in_function_def (self : mapper)
      the attribute to the whole expression, in general, when shuffuling the ast
      it is very hard to place attributes correctly
   *)
+  | Pexp_letmodule
+      (lid, ({pmod_desc = Pmod_ident {txt}; pmod_attributes} as me), expr)
```

Comment:
Do you need this too, or are the use cases of this kind more rare?

```ml
  | Pmod_constraint of module_expr * module_type
        (* (ME : MT) *)
```

Reply:
Oops, correct! I'll add it.

Reply:
https://github.com/rescript-lang/rescript-compiler/pull/6310/commits/da381bab81c5c4bdb01e702c3fda6b3bcfec71b0

I've added the case of `Pmod_constraint ({pmod_desc = Pmod_ident _}, {pmty_desc = Pmty_ident mtyp_lid})` additionally, e.g. `ME: MT`. In other cases, for example, I don't think `Pmty_alias` or `Pmty_with are applicable` for dynamic import. Any thought?